### PR TITLE
system/trace: correct arg index if command is alreay set

### PR DESCRIPTION
**Pull request by anchao**
## Summary

1. trace: format the usage output

before:
```
Usage: trace <subcommand>...
Subcommand:
  start [-c][<duration>]          : Start task tracing
  stop                            : Stop task tracing
  dump [-a][-c][<filename>]           : Output the trace result
 [-a] <Android SysTrace>
  mode [{+|-}{o|w|s|a|i|d}...]        : Set task trace options
  switch [+|-] : Configure switch trace filter
  syscall [{+|-}<syscallname>...] : Configure syscall trace filter
  irq [{+|-}<irqnum>...]          : Configure IRQ trace filter
  print [+|-] : Configure dump tnsh>
```

after:
```
Usage: trace <subcommand>...
Subcommand:
        start [-c][<duration>]          : Start task tracing
        stop                            : Stop task tracing
        dump [-a][-c][<filename>]       : Output the trace result
                                          [-a] <Android SysTrace>
        mode [{+|-}{o|w|s|a|i|d}...]    : Set task trace options
        switch [+|-]                    : Configure switch trace filter
        syscall [{+|-}<syscallname>...] : Configure syscall trace filter
        irq [{+|-}<irqnum>...]          : Configure IRQ trace filter
        print [+|-]                     : Configure dump trace filter
```

2. system/trace: correct arg index if command is alreay set

An error occurs when the same command is executed twice

```
$ trace switch +
$ trace switch +
Usage: trace <subcommand>...
Subcommand:
  start [-c][<duration>]          : Start task tracing
  stop                            : Stop task tracing
  dump [-a][-c][<filename>]           : Output the trace result
 [-a] <Android SysTrace>
  mode [{+|-}{o|w|s|a|i|d}...]        : Set task trace options
  switch [+|-] : Configure switch trace filter
  syscall [{+|-}<syscallname>...] : Configure syscall trace filter
  irq [{+|-}<irqnum>...]          : Configure IRQ trace filter
  print [+|-] : Configure dump tnsh>
```

3. system/sched_note: fix build break

Signed-off-by: chao.an <anchao@xiaomi.com>

## Impact

N/A

## Testing

ci-check

### CODE REVIEW

File: system/trace/trace.c [hunk 1]
```diff
@@ -755,12 +756,15 @@ static int trace_cmd_print(int index, int argc, FAR char **argv,
       if (argv[index][0] == '-' || argv[index][0] == '+')
         {
           enable = (argv[index][0] == '+');
+
+          index++;
+
```
Comment (pkarashchenko) [inline system/trace/trace.c:761]: Optional
```suggestion
          enable = (argv[index++][0] == '+');
```

File: system/trace/trace.c [hunk 2]
```diff
@@ -502,12 +502,15 @@ static int trace_cmd_switch(int index, int argc, FAR char **argv,
...
```
Comment (pkarashchenko) [inline system/trace/trace.c:507]: Optional
```suggestion
          enable = (argv[index++][0] == '+');
```

File: system/trace/trace.c [hunk 3]
```diff
@@ -793,38 +793,40 @@ static void show_usage(void)
   fprintf(stderr,
           "\nUsage: trace <subcommand>...\n"
           "Subcommand:\n"
-          "  start [-c][<duration>]          :"
+          "\tstart [-c][<duration>]\t\t:"
```
[the remark this answers is not in the document]
Comment (pkarashchenko) [reply system/trace/trace.c:796]: Yeah. Spaces seems to be more portable.

[the remark this answers is not in the document]
Comment (anchao) [reply system/trace/trace.c]: Done

File: system/trace/trace.c (hunk 2 above)
Comment (anchao) [reply system/trace/trace.c]: Done

File: system/trace/trace.c (hunk 1 above)
Comment (anchao) [reply system/trace/trace.c]: Done

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**